app/models/book.rb

```diff
@@ -0,0 +1,23 @@
+class Book < ApplicationRecord
+  has_one_attached :cover
+  has_many_attached :images
+
+  has_many :book_authors, dependent: :destroy
+  has_many :authors, through: :book_authors
+  has_many :book_categories, dependent: :destroy
+  has_many :categories, through: :book_categories
+  has_many :order_books, dependent: :destroy
+  has_many :orders, through: :order_books
+  has_many :reviews, dependent: :destroy
+
+  scope :by_filter, -> (filter) { public_send(filter) }
```

Comment:
delete and destroy are public methods too

redone it with check for concrete filter names